pywatts/wrapper/sm_time_series_model_wrapper.py

```diff
@@ -0,0 +1,170 @@
+from pywatts.wrapper.base_wrapper import BaseWrapper
+
+
+class SmTimeSeriesModelWrapper(BaseWrapper):
+    """
+    Wrapper for statsmodels modules.
+
+    :param module: The statsmodels module to wrap. Not this module should not be initialised.
+    :param name: The name of the module
+    :type name: str
+    :param module_kwargs: The module keyword arguments necessary for creating the statsmodel module
+    :type module_kwargs: dict
+    :param fit_kwargs: The optional fit keyword arguments for fitting the model
+    :type fit_kwargs: dict
+    :param predict_kwargs: The optional predict keyword arguments for predicting with the model (except start and end)
+    :type predict_kwargs: dict
+    """
+
+    def __init__(self, module: Type[TimeSeriesModel], name: str = None, module_kwargs=None,
+                 fit_kwargs=None, predict_kwargs=None):
+        if name is None:
+            name = module.__name__
+        super().__init__(name)
+        if fit_kwargs is None:
+            fit_kwargs = {}
+        if predict_kwargs is None:
+            predict_kwargs = {}
+        self.module = module
+        if not module_kwargs:
+            module_kwargs = {}
+        self.module_kwargs = module_kwargs
+        self.fit_kwargs = fit_kwargs
+        self.predict_kwargs = predict_kwargs
+
+    def get_params(self) -> Dict[str, object]:
+        """
+        Returns the parameters of the statsmodels module.
+
+        :return: A dict containing the module keyword arguments, the fit keyword arguments, the predict keyword
+        arguments and the fitted model parameters
+        :rtype: Dict
+        """
+        return {
+            "module_kwargs": self.module_kwargs,
+            "fit_kwargs": self.fit_kwargs,
+            "predict_kwargs": self.predict_kwargs,
+        }
+
+    def set_params(self, module_kwargs=None, fit_kwargs=None, predict_kwargs=None):
+        """
+        Set the parameters of the statsmodels wrapper
+
+        :param module_kwargs: keyword arguments for the statsmodel module.
+        :type module_kwargs: Dict
+        :param fit_kwargs: keyword arguments for the fit method.
+        :type fit_kwargs: Dict
+        :param predict_kwargs: keyword arguments for the predict method.
+        :type predict_kwargs: Dict
+        """
+        if module_kwargs:
+            self.module_kwargs = module_kwargs
+        if fit_kwargs:
+            self.fit_kwargs = fit_kwargs
+        if predict_kwargs:
+            self.predict_kwargs = predict_kwargs
+
+    def fit(self, **kwargs: xr.DataArray):
+        """
+        Fits the statsmodels module
+
+        :param kwargs: A dict of input arrays
+        :type kwargs: xr.DataArray
+        """
+        x = []
+        y = []
+        for key, value in kwargs.items():
+            if key.startswith("target"):
+                y.append(value.values.reshape(-1))
+            else:
+                x.append(value.values)
```

Comment:
Redundant code. Isn't it possible that pywatts will parse target attributes? Maybe by default or by offering a `x, y = self._get_targets(kwargs)` or `_split_targets` or `_parse_targets` ...

Reply:
Done